hs/src/Reach/Simulator/Core.hs

```diff
@@ -276,25 +300,33 @@ instance Interp DLExpr where
         -- TODO: "Remove funds from contract..." which contract?
         (V_UInt n, V_Address acc) -> do
           case maybe_dlarg of
-            Nothing -> updateLedger acc 0 (+n) $ return V_Null
+            Nothing -> do
+              updateLedger acc 0 (+n)
```

Comment:
It is weird that `updateLedger` is taking a `Num -> Num` function, because it needs to know the amount itself so it can remove from one and give to the other

Reply:
it gets called once for each of those